### PR TITLE
feat: convert lib to be synchronous

### PR DESCRIPTION
**Pull request by gratcliff**
| 🚥 Resolves ISSUE_ID |
| :------------------- |

## 🧰 Changes
After some thought, input, (and compilation errors due to ESM), I've decided to rewrite this library from the ground up. 

We have no documented all existing headers as they exist in MDN, complete with `markdown`, links, and `deprecated` + `experimental` flags. This should much better fit our use cases, as well as be holistically easier to use.

## 🧬 QA & Testing

Provide as much information as you can on how to test what you've done.

### CODE REVIEW

**Comment by gratcliff** [comment]
Updating `README.md` with updated usage guide.

File: src/http-headers.ts [hunk 1]
```diff
@@ -0,0 +1,728 @@
+import type { HTTPHeader, HTTPHeaderDescription } from './types';
+
+const HTTPHeaders: Record<HTTPHeader, HTTPHeaderDescription> = {
+  Accept: {
+    description: 'Informs the server about the types of data that can be sent back.',
+    link: 'https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Accept',
+    markdown: 'Informs the server about the {{Glossary("MIME_type", "types")}} of data that can be sent back.',
```
Comment (erunion) [inline src/http-headers.ts:7]: Probably want to scrub these template functions.

File: src/http-headers.ts [hunk 2]
```diff
@@ -0,0 +1,728 @@
...
+  },
+  Connection: {
+    description: 'Controls whether the network connection stays open after the current transaction finishes.',
+    link: 'https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Connection',
+  },
+  Cookie: {
+    description: 'Contains stored HTTP cookies previously sent by the server with the Set-Cookie header.',
+    link: 'https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Cookie',
+    markdown:
+      'Contains stored [HTTP cookies](/en-US/docs/Web/HTTP/Cookies) previously sent by the server with the {{HTTPHeader("Set-Cookie")}} header.',
```
Comment (erunion) [inline src/http-headers.ts:130]: Another one

**Review by erunion — APPROVED** [review]
love how this thing has no deps anymore. hell yea

**Pull request merged**